rust/azure_iot_operations_connector/examples/base_connector_sample.rs

```diff
@@ -351,6 +485,42 @@ pub fn mock_received_data(count: u32) -> Data {
     }
 }
 
+/// Only reports status on first time (None) or when changing from OK to Error.
+/// Skips reporting when status has already been reported and hasn't changed.
+fn report_status_if_changed(
```

Comment:
Technically this function is a factory for a function that does "report_status_if_changed" - it isn't itself the function/closure.

Not necessarily wrong, but I wonder if there's another way you could do this with function pointers, although it does look like you want to pull in the `log_identifier` so maybe this is ideal. Regardless, I do think you should frame it as a factory if you do keep this pattern.